provider/datagen/src/registry.rs

```diff
@@ -404,6 +404,66 @@ registry!(
     icu_timezone::provider::MetazonePeriodV1Marker = "time_zone/metazone_period@1",
 );
 
+pub use heap_measurements::MeasuringAllocator;
+
+// Inspired by the assert_no_alloc crate
+mod heap_measurements {
```

Comment:
thought: not a huge fan of this living in the registry, and also not a fan of this living in the datagen lib in the first place: ought it be moved to a test submodule?

Reply:
It can't because `buf.into_deserialized` needs to be measured, but `icu_provider::dynutil::UpcastDataPayload::upcast` cannot, as that allocates a `Box<dyn ...>`.

Reply:
right but we can keep the old api with the closure, right? Just make it return two things